### PR TITLE
rename `reqs` arg to `connection_options` (#197)

### PR DESCRIPTION
**Pull request by AdamWill**
As discussed in #197, this renames the `reqs` argument to `Site` to `connection_options`. The old name still works but triggers a DeprecationWarning. If you specify both we raise ValueError.

### CODE REVIEW

**Comment by AdamWill** [comment]
Any thoughts on this before I merge it? @mwclient/maintainers

**Comment by RheingoldRiver** [comment]
LGTM

**Review by marcfrederick — APPROVED** [review]
LGTM, thank you!

**Pull request merged**
